fireplace/dsl/random_picker.py

```diff
@@ -1,79 +1,97 @@
 import random
 from hearthstone.enums import CardType, Race, Rarity
 from .lazynum import LazyValue
+from copy import copy
+from .. import cards
 
 
 class RandomCardPicker(LazyValue):
 	"""
 	Store filters and generate a random card matching the filters on pick()
+	Constructor takes a single global set of filters and uses a default weighting of 1
+	Additional weighted filter sets can be added with add(), these will be merged with the global filters
 	"""
-	def __init__(self, *args, **filters):
-		self.args = args
-		self.filters = filters
+	def __init__(self, **filters):
+		self.weights = []
+		self.weightedfilters = []
+		self.globalfilter = filters
 		self.count = 1
-		self._cards = None
-		self.lazy_filters = False
-		for v in filters.values():
-			if isinstance(v, LazyValue):
-				self.lazy_filters = True
-				break
 
 	def __repr__(self):
-		return "%s(%r)" % (self.__class__.__name__, self.filters)
+		return "%s(%r)" % (self.__class__.__name__, self.globalfilter)
 
+	# select number of cards to fetch
 	def __mul__(self, other):
-		ret = self.__class__(*self.args, **self.filters)
+		ret = copy(self)
 		ret.count = other
 		return ret
 
-	@property
-	def cards(self):
```

Comment:
A few reasons:

1. Clarity. There were previously 3 methods that found cards in different ways and via different codepaths. Now there is 1.
2. If a filter was a Lazy filter the cards property would not return the correct values
3. The property is a read-only property anyway
4. find_cards allows you (or a subclass) to specify an optional filter set each time and get a new set of cards back each time. Lazy filters will also work. This can be re-used as much as desired. If no filter is specified, the global filter set is used. You can't do this with a property.
5. Sub-classes can supply their own card lists in the new 'cards' argument to evaluate() if they want to override the list